.bit/responses/2.4-Week 2 Step 4.md

```diff
@@ -1,53 +1,97 @@
+
+*What is happening in this flowchart?*
+
+**Azure Function:** We are now going to be creating parameters that the Azure Function will use to send a request to the Face API Endpoint we just created. 
+
+### Concepts to know:
+* GET and POST requests
+* API Endpoints
+* Async Functions
+* Parameters
+
+In this section, we'll be focusing on Part 2.
+
+At this point, your Azure function should look like this:
+
+```js
+var multipart = require("parse-multipart");
+  
+module.exports = async function (context, req) {
+    context.log('JavaScript HTTP trigger function processed a request.'); 
+
+    var boundary = multipart.getBoundary(req.headers['content-type']);
+    
+    var body = req.body;
+  
+    var parts = multipart.Parse(body, boundary);
+};
+```
+
+
+### 📝 Task 4: Create a new function and set the parameters of your POST request.
+<details>
+<summary>❓Where do I begin?</summary>
+</br>
+
+Create a new function, outside of  `module.exports`  that will handle analyzing the image (this function is **async** because we will be using the **await** keyword with the API call).
+
+This function will be called `analyzeImage(img)` and takes in one parameter, `img`, that contains the image we're trying to analyze.  Inside, we have two variables involved in the call: `subscriptionKey`  and `uriBase`.  Substitute the necessary values with your own info.
+
+```js
+async function analyzeImage(img){
+    const subscriptionKey = '<YOUR SUBSCRIPTION KEY>';
+    const uriBase = '<YOUR ENDPOINT>' + '/face/v1.0/detect';
+}
+```
+
+</details>
+<br>
+
+<details>
+<summary>❓What are the parameters for the request?</summary>
+</br>
+The documentation for the Face API is here: https://westus.dev.cognitive.microsoft.com/docs/services/563879b61984550e40cbbe8d/operations/563879b61984550f30395236. Read through it, and notice that the request url is this:
+
+`https://{endpoint}/face/v1.0/detect\[?returnFaceId]\[&returnFaceLandmarks]\[&returnFaceAttributes]\[&recognitionModel]\[&returnRecognitionModel][&detectionModel]`
+
+All of the bracketed sections represent possible request parameters. Read through **Request Parameters** section carefully. How can we specify that we want to get the emotion data?
+
+</details>
+<br>
+
+<details>
+<summary>❓How do I specify parameters?</summary>
+</br>
+In order to specify all of our parameters easily, we're going to create a new `URLSearchParams`  object. Here's the object declared for you. I've also already specified one parameter, `returnFaceId`,  as `true` to provide an example. Add in a new parameter that requests emotion.
+
+```js
+let params = new URLSearchParams({
+	'returnFaceId': 'true',
+	'<PARAMETER NAME>': '<PARAMETER VALUE>'     //FILL IN THIS LINE
+})
+```
+
+</details>
+<br>
+
+
+**To move on, comment the code in your Azure Function below. BUT DO NOT INCLUDE YOUR SUBSCRIPTION KEY. DELETE IT, and then copy your code in.**

```

Comment:
It would be great if we could change the format to the h2 titles for "move on"; here's the format again:

## :camping: To move on...

Also, if you know how to use environment variables in Azure, it would be awesome if you could write a quick drop down on how to use them and edit the code to use `process.env.[KEY NAME]`